docs/api/api.md

```diff
@@ -1,30 +1,41 @@
 # Full API list
 <!--these are now just the APIs - do I want to add the tutorials?-->
 
+This document lists Mbed OS components supported APIs. In general, it lists both common APIs (RTOS and the bare metal profile supported), and RTOS only supported APIs, but occasionally, It has either one.
+
 ## RTOS and event handling
 
 The Mbed OS RTOS capabilities include managing objects such as threads, synchronization objects and timers. It also provides interfaces for attaching an application-specific idle hook function, reads the OS tick count and implements functionality to report RTOS errors.
 
+Most of the Mbed OS APIs are also supported with the Mbed OS bare metal profile.
 
 <table>
 <tbody>
 <tr>
-<td><a href="rtos-apis.html">Thread</a></td>
-<td><a href="thisthread.html">ThisThread</a></td>
-<td><a href="mutex.html">Mutex</a></td>
 <td><a href="semaphore.html">Semaphore</a></td>
+<td><a href="userallocatedevent.html">UserAllocatedEvent</a></td>
+<td><a href="eventflags.html">EventFlags</a></td>
+<td><a href="kernel-interface-functions.html">Kernel interface functions</a></td>
```

Comment:
Only get_ms_count is supported in bare metal.
attach_idle_hook & attach_thread_terminate_hook are RTOS only functions.
I think it is better to say that "Kernel interface functions" are RTOS only except for one. Note that you will need doxygen for Kernel.h

Reply:
We could add this information in the `kernel-interface-functions.html` file instead of describe in api.md file.
@iriark01 Could you please suggest